map_gen/maps/danger_ores/modules/expanse.lua

```diff
@@ -0,0 +1,501 @@
+local b = require 'map_gen.shared.builders'
+local Event = require 'utils.event'
+local Generate = require 'map_gen.shared.generate'
+local Global = require 'utils.global'
+local Price_raffle = require 'map_gen.maps.danger_ores.modules.price_raffle'
+local RS = require 'map_gen.shared.redmew_surface'
+local table = require 'utils.table'
+local Task = require 'utils.task'
+local Token = require 'utils.token'
+
+local rendering = rendering
+local math_abs = math.abs
+
+local rock_resources = {
+  'iron-ore', 'iron-ore', 'iron-ore', 'iron-ore', 'iron-ore',
+  'copper-ore', 'copper-ore', 'copper-ore',
+  'coal'
+}
+local price_modifiers = {
+  ['cliff'] = -128,
+  ['deepwater-green'] = -5,
+  ['deepwater'] = -5,
+  ['simple-entity'] = 2,
+  ['tree'] = -8,
+  ['turret'] = -128,
+  ['unit-spawner'] = -256,
+  ['unit'] = -16,
+  ['water-green'] = -5,
+  ['water-mud'] = -6,
+  ['water-shallow'] = -6,
+  ['water'] = -5,
+}
+
+---@param config
+---@field start_size              int, the width/height of the starting area. Will be ceiled to a multiple of a chunk
+---@field chance_to_receive_token int, probability of coin rewards [0, 1]
+---@field max_ore_price_modifier  int, max value for the ore_modifier after cmputing it with the distance_modifier
+---@field price_distance_modifier int, how distance from {0,0} influence the request price
+return function(config)
+  Generate.enable_register_events = false
+
+  local cell_size = 32
+  local start_size = config.start_size
+  local chance_to_receive_token = config.chance_to_receive_token or 0.20
+  local max_ore_price_modifier  = config.max_ore_price_modifier  or 0.33
+  local price_distance_modifier = config.price_distance_modifier or 0.006
+
+  -- pad start_size to closest multiple of 32 for chunk alignment
+  if start_size % cell_size ~= 0 then
+    start_size = start_size + cell_size - start_size % cell_size
+  end
+
+  local surface
+  local primitives = { index = nil }
+  local chest_data = {}
+  local grid = {}
+
+  local bounds = b.rectangle(start_size, start_size)
+
+  local function on_chunk_generated(event)
+    if surface ~= event.surface then
+      return
+    end
+
+    local left_top = event.area.left_top
+    local x, y = left_top.x, left_top.y
+
+    if bounds(x + 0.5, y + 0.5) then
+      -- Starting area
+      Generate.do_chunk(event)
+    else
+      -- Outside starting area
+      local tiles = {}
+      for x1 = x, x + 31 do
+        for y1 = y, y + 31 do
+          tiles[#tiles + 1] = { name = 'out-of-map', position = { x1, y1 } }
+        end
+      end
+      surface.set_tiles(tiles, true)
+    end
+  end
+
+  local function reward_tokens(entity)
+    local chance = chance_to_receive_token % 1
+    local count = math.floor(chance_to_receive_token)
+
+    if chance > 0 then
+      chance = math.floor(chance * 1000)
+      if math.random(1, 1000) <= chance then
+        entity.surface.spill_item_stack(entity.position, { name = 'coin', count = 1 }, true, nil, false)
+      end
```

Comment:
Just a suggestion, but wouldn't this be simpler?

```suggestion
      if math.random() < chance then
        entity.surface.spill_item_stack(entity.position, { name = 'coin', count = 1 }, true, nil, false)
      end
```

Note: `math.random()` without args is [0,1), so the change in comparison operator was deliberate.

Reply:
Yes. Honestly this part is from the batch of code I blindly copied over and never even looked at since it never caused any issue. But now that I give it a proper look, since I define the chance token to be [0,1], I don't even need to do the modulo operation, allocate count, and check for count > 0. Just `math.random() < token_chance` and get rid of all other 15 lines of bloat